### PR TITLE
Update WTS multiqc config value

### PR DESCRIPTION
**Pull request by skanwal**
Resolves https://github.com/umccr/infrastructure/issues/277, using Dragen-v3.9.3

## TO-DO

- [x] Wait for `wfr.6cc0fc8a02c14310824a7c4500dfa768` to finish successfully
- [x] Compare multiQC report

### CODE REVIEW

**Comment by victorskl** [comment]
- LGTM. But. I will wait for ready review flag.
- Queued into `2023.01` release milestone.

**Comment by alexiswl** [comment]
As discussed with @skanwal yesterday, we need to extend the list of sample_names to include the library names (along with the sample ids). This is because the qualimap multiqc module uses the directory name, but the qualimap rnaseq qc module uses the name of the bam file (which is the sample id).  :/ 

We can rename with the keys <library_name>_dragen_qualimap but have decided that the new qualimap reference directories should use <library_name>_qualimap instead which reflects the current output in the production workflows. If we do upgrade to Dragen v4 we'll take that approach (since the references will need to be re-run anyway).

**Comment by skanwal** [comment]
MultiQC report output from wfr.d5b3c175e42048bba2af3cd91a02e225 looks good to me.

For record: Inputs used were:

```
{
    "name": "Dragen-3.9.3-QC-Reference-Samples-Config-Update",
    "input": {
        "annotation_file": {
            "class": "File",
            "location": "gds://development/reference-data/dragen_wts/hg38/ref-transcripts.non-zero-length.gtf"
        },
        "reference_tar": {
            "class": "File",
            "location": "gds://development/reference-data/dragen_hash_tables/v8/hg38/altaware-cnv-anchored/hg38-v8-altaware-cnv-anchored.tar.gz"
        },
        "protein_domains": {
            "class": "File",
            "location": "gds://development/reference-data/dragen_wts/arriba/hg38/arriba-protein-domains.gff3"
        },
        "enable_map_align_output": true,
        "enable_duplicate_marking": false,
        "enable_sort": true,
        "output_file_prefix": "MDX210377",
        "output_directory": "L2101522_dragen",
        "fastq_list_rows": [
            { 
                "rgid": "MDX210377",
                "rgsm": "MDX210377",
                "rglb": "L2101522",
                "lane": 1,
                "read_1": {
                  "class": "File",
                  "location": "gds://production/primary_data/211125_A00130_0185_AHWC2HDSX2/2022010575cdcf71/WTS_NebRNA/MDX210377_L2101522_S2_L004_R1_001.fastq.gz"
                },
                "read_2": {
                  "class": "File",
                  "location": "gds://production/primary_data/211125_A00130_0185_AHWC2HDSX2/2022010575cdcf71/WTS_NebRNA/MDX210377_L2101522_S2_L004_R2_001.fastq.gz"
                }
            }
        ],
        "cytobands": {
            "class": "File",
            "location": "gds://development/reference-data/dragen_wts/arriba/2.3.0/cytobands_hg38_GRCh38_v2.3.0.tsv"
        },
        "blacklist": {
            "class": "File",
            "location": "gds://development/reference-data/dragen_wts/arriba/2.3.0/blacklist_hg38_GRCh38_v2.3.0.tsv.gz"
        },
        "reference_fasta": {
            "class": "File",
            "location": "gds://development/reference-data/genomes/hg38/hg38.fa"
        },
        "qc_reference_samples": [
            {
              "class": "Directory",
              "location": "gds://development/reference-data/dragen_wts/multiqc/2022-09-14--3.9.3/SBJ01563"
            },
            {
              "class": "Directory",
              "location": "gds://development/reference-data/dragen_wts/multiqc/2022-09-14--3.9.3/SBJ01147"
            },
            {
              "class": "Directory",
              "location": "gds://development/reference-data/dragen_wts/multiqc/2022-09-14--3.9.3/SBJ01620"
            },
            {
              "class": "Directory",
              "location": "gds://development/reference-data/dragen_wts/multiqc/2022-09-14--3.9.3/SBJ01286"
            },
            {
              "class": "Directory",
              "location": "gds://development/reference-data/dragen_wts/multiqc/2022-09-14--3.9.3/SBJ01673"
            }
          ],
          "cl_config": "{\"sample_names_replace\":{\"PRJ220412\":\"Ref_1_Good\",\"MDX210318\":\"Ref_2_Good\",\"PRJ220466\":\"Ref_3_Good\",\"PRJ211234\":\"Ref_4_Bad\",\"PRJ220790\":\"Ref_5_Bad\",\"L2200121_dragen_qualimap\":\"Ref_1_Good\",\"L2101521_dragen_qualimap\":\"Ref_2_Good\",\"L2200188_dragen_qualimap\":\"Ref_3_Good\",\"L2101763_dragen_qualimap\":\"Ref_4_Bad\",\"L2200311_dragen_qualimap\":\"Ref_5_Bad\",}}"
    },
    "engineParameters": {
        "outputDirectory": "gds://development/temp/sehrish/dragne_wts_debug_qc/MDX210377_rerun",
        "workDirectory": "gds://development/temp/sehrish/dragne_wts_debug_qc/MDX210377__rerun_work"
    }
}
```

**Comment by skanwal** [comment]
@alexiswl - updated in [15b78c4](https://github.com/umccr/infrastructure/pull/280/commits/15b78c4f3702ebff04084d8a6f772284e0f05aa1) and is ready for review.

File: terraform/stacks/umccr_data_portal/workflow/template/wts_tumor_only_dev.json
Comment (alexiswl) [inline terraform/stacks/umccr_data_portal/workflow/template/wts_tumor_only_dev.json:53]: Extra comma at the end here, will need to remove it

```
"Ref_5_Bad\",}}"
```

should be

```
"Ref_5_Bad\"}}"
```

```bash
jq '' <<< "{\"sample_names_replace\":{\"PRJ220412\":\"Ref_1_Good\",\"MDX210318\":\"Ref_2_Good\",\"PRJ220466\":\"Ref_3_Good\",\"PRJ211234\":\"Ref_4_Bad\",\"PRJ220790\":\"Ref_5_Bad\",\"L2200121_dragen_qualimap\":\"Ref_1_Good\",\"L2101521_dragen_qualimap\":\"Ref_2_Good\",\"L2200188_dragen_qualimap\":\"Ref_3_Good\",\"L2101763_dragen_qualimap\":\"Ref_4_Bad\",\"L2200311_dragen_qualimap\":\"Ref_5_Bad\",}}"
parse error: Expected another key-value pair at line 1, column 347
```

But 

```
 jq '' <<< "{\"sample_names_replace\":{\"PRJ220412\":\"Ref_1_Good\",\"MDX210318\":\"Ref_2_Good\",\"PRJ220466\":\"Ref_3_Good\",\"PRJ211234\":\"Ref_4_Bad\",\"PRJ220790\":\"Ref_5_Bad\",\"L2200121_dragen_qualimap\":\"Ref_1_Good\",\"L2101521_dragen_qualimap\":\"Ref_2_Good\",\"L2200188_dragen_qualimap\":\"Ref_3_Good\",\"L2101763_dragen_qualimap\":\"Ref_4_Bad\",\"L2200311_dragen_qualimap\":\"Ref_5_Bad\"}}"
```

Yields

```
{
  "sample_names_replace": {
    "PRJ220412": "Ref_1_Good",
    "MDX210318": "Ref_2_Good",
    "PRJ220466": "Ref_3_Good",
    "PRJ211234": "Ref_4_Bad",
    "PRJ220790": "Ref_5_Bad",
    "L2200121_dragen_qualimap": "Ref_1_Good",
    "L2101521_dragen_qualimap": "Ref_2_Good",
    "L2200188_dragen_qualimap": "Ref_3_Good",
    "L2101763_dragen_qualimap": "Ref_4_Bad",
    "L2200311_dragen_qualimap": "Ref_5_Bad"
  }
}
```

File: terraform/stacks/umccr_data_portal/workflow/template/wts_tumor_only_prod.json
Comment (alexiswl) [inline terraform/stacks/umccr_data_portal/workflow/template/wts_tumor_only_prod.json:53]: See [`15b78c4` (#280)](https://github.com/umccr/infrastructure/pull/280/commits/15b78c4f3702ebff04084d8a6f772284e0f05aa1#r1107825065)

File: terraform/stacks/umccr_data_portal/workflow/template/wts_tumor_only_stg.json
Comment (alexiswl) [inline terraform/stacks/umccr_data_portal/workflow/template/wts_tumor_only_stg.json:53]: See [`15b78c4` (#280)](https://github.com/umccr/infrastructure/pull/280/commits/15b78c4f3702ebff04084d8a6f772284e0f05aa1#r1107825065)

File: terraform/stacks/umccr_data_portal/workflow/template/wts_tumor_only_dev.json
Comment (skanwal) [reply terraform/stacks/umccr_data_portal/workflow/template/wts_tumor_only_dev.json]: updated in https://github.com/umccr/infrastructure/commit/3b6c25b16e359aa95e5e9ad540b0ef32709e5516

File: terraform/stacks/umccr_data_portal/workflow/template/wts_tumor_only_prod.json
Comment (skanwal) [reply terraform/stacks/umccr_data_portal/workflow/template/wts_tumor_only_prod.json]: updated in https://github.com/umccr/infrastructure/commit/3b6c25b16e359aa95e5e9ad540b0ef32709e5516

File: terraform/stacks/umccr_data_portal/workflow/template/wts_tumor_only_stg.json
Comment (skanwal) [reply terraform/stacks/umccr_data_portal/workflow/template/wts_tumor_only_stg.json]: updated in https://github.com/umccr/infrastructure/commit/3b6c25b16e359aa95e5e9ad540b0ef32709e5516

**Comment by alexiswl** [comment]
Thanks @skanwal, do we also want to update the workflow version to include this feature? https://github.com/umccr/cwl-ica/pull/281

**Comment by alexiswl** [comment]
> Thanks @skanwal, do we also want to update the workflow version to include this feature? https://github.com/umccr/cwl-ica/pull/281

@skanwal following this up

**Comment by skanwal** [comment]
@alexiswl - oh yes, definitely. Think we should also run a  workflow test with updated version as well.

**Comment by skanwal** [comment]
@alexiswl I have updated WTS prod workflow version in [eb82d26](https://github.com/umccr/infrastructure/pull/280/commits/eb82d26de7ca2779dd2503d5497bf50330fbf8e6). Do we also need to update one for `stg`?

File: terraform/stacks/umccr_data_portal/workflow/wts_tumor_only.tf
Comment (alexiswl) [inline terraform/stacks/umccr_data_portal/workflow/wts_tumor_only.tf:10]: I don't think this version ID is correct, PR https://github.com/umccr/cwl-ica/pull/284 created the production version (3.9.3--8fdabfc) for the dragen transcriptome pipeline (see https://github.com/umccr/cwl-ica/pull/284/files#diff-b47734da5a8e4c08b51d28d2b39f874c872377c258d5fecc1d5072c93b2572dfR1805-R1809) for more info. 

Can also run the following to validate
```
ica workflows versions list wfl.7e5ba7470b5549a6b4bf6d95daaa1214 -o json | \
jq --raw-output \
  '
    .items | 
    sort_by(.timeModified) | 
    reverse | 
    map(.version)[0:5]
  '
```

Yields

```
[
  "3.9.3--8fdabfc",
  "3.7.5--8fdabfc",
  "3.9.3--a0960f3",
  "3.9.3--a8fdcf1",
  "3.9.3--3b07571"
]
```

**Comment by alexiswl** [comment]
> Do we also need to update one for stg?

Yes, staging should also be updated

File: terraform/stacks/umccr_data_portal/workflow/wts_tumor_only.tf
Comment (alexiswl) [reply terraform/stacks/umccr_data_portal/workflow/wts_tumor_only.tf:10]: Correction: this does contain https://github.com/umccr/cwl-ica/pull/281 but I'd still update to https://github.com/umccr/cwl-ica/pull/284 to ensure if we do want to move to using presigned urls at some point.

Comment (skanwal) [reply terraform/stacks/umccr_data_portal/workflow/wts_tumor_only.tf]: Okay, as noted I had pointed to https://github.com/umccr/cwl-ica/pull/281. Now updated to include https://github.com/umccr/cwl-ica/pull/284 in https://github.com/umccr/infrastructure/commit/fe2a41e8aca050fe8a2eb637346ee79ce41cc70f

**Comment by alexiswl** [comment]
Great! Can you tick-off 'Compare multiQC report' and I'll approve

**Comment by skanwal** [comment]
Done.

**Review by alexiswl — APPROVED** [review]

**Review by victorskl — APPROVED** [review]

**Comment by victorskl** [comment]
Thanks both. LGTM for merge... I will apply changes when release.

**Comment by victorskl** [comment]
Technically LGTM. @ohofmann Could you pls review and, tick approve if all good.

**Review by ohofmann — APPROVED** [review]
Thanks for fielding this!

**Comment by victorskl** [comment]
Thanks all; merging...

**Pull request merged**
